### DISCUSSION TITLE
Giving nuclei count from segmentation for deconvolution

### DISCUSSION THREAD

**Discussion by HugoCroizer** [BayraktarLab/cell2location#70] [Ideas] [answered]
Hello,

Thank you for the really nice package, and the new tutorial which I'm going to try soon. 

From my first test using cell2location, I see that the relative cell abundance per spot is nicely correlated with the segmented nuclei counts. However the estimation are a bit low, when I see a mean of 11 cells per spots, the deconvolution sees only a mean of 7 cells. However the distribution is really quite the same.

I'm going to try to increase the cells_per_spot parameter, but I would also like to give the nuclei count per spot from the segmentation. But I can't find a way to do it.

Is it possible ? 
Thank you

**Comment by vitkl** [comment, marked as answer]
Hi @Dantferno 

Thanks for using cell2location and it is really nice to see that you can also validate that the **absolute** cell abundance per location is nicely correlated with the segmented nuclei counts.

I would encourage you to try scvi-tools/pyro-based version to see if it gives better consistency (if decomposition sees 11 cells per location) because we observed that the new version is more numerically accurate. Importantly, the new version also employs per-location normalisation to account for technical differences in RNA detection. Both of these factors can improve estimation.

While it is possible to provide segmentation based numbers to the old version (simply provide a numpy array (location * 1) instead of a single number) - we did not see any accuracy benefits from including that information in the mouse brain data. This feature is not available in the new pyro-based version for now.

Importantly, segmentation based count also doesn't account for the fact that **spatial txn technologies capture RNA from partial cells**. So while you see 11 nuclei in the histology image, real cell abundance after accounting for thin 2D sectioning and partial overlap with spots could be 7.35. So correspondence between nuclei number in H&E and cell abundance in txn data could be less straightforward.

In summary, I would encourage trying the new version of the package before investigating how to provide segmentation based counts.
